### PR TITLE
last.fm code improvements

### PR DESCRIPTION
**Pull request by yourordinarycat**
**Resolved / Related Issues**
This PR improves the code for last.fm handling.

**Details of Changes**
- Added a new LastFMViewModel
- Added helper method to get encoded hashes from strings
- Cleaned up methods, switched to the WinRT Cryptography and Http classes

**Validation**
Ran the app, logged in through the OnlineServicesPage.

**Screenshots / Videos (optional)**
None.

### CODE REVIEW

**Review by itsWindows11 — APPROVED** [review]

**Review by itsWindows11 — CHANGES_REQUESTED** [review]
Overall it works, but when last.fm reports that it can't connect to the app, it seems to crash with a FileNotFoundException in TryAuthenticateAsync

**Comment by yourordinarycat** [comment]
How do you recreate that? Wrong credentials or something?

**Comment by itsWindows11** [comment]
> How do you recreate that? Wrong credentials or something?

It might happen in some cases after logging in then closing iirc but I wasn't able to reproduce the last.fm app connection issue again, but we might need to handle this.

**Comment by yourordinarycat** [comment]
Seems to be a host unavailable/no internet access thing. Should be fixed now.

File: Rise Media Player Dev/Settings/OnlineServicesPage.xaml.cs [hunk 1]
```diff
@@ -13,30 +13,23 @@ namespace Rise.App.Settings
...
-                LastFMStatus.IsEnabled = false;
-            }
-            catch (Exception e1)
-            {
-                Debug.WriteLine(e1.Message);
-            }
+            if (!WebHelpers.IsInternetAccessAvailable())
+                return;
+
+            bool result = await ViewModel.TryAuthenticateAsync();
```
Comment (itsWindows11) [inline Rise Media Player Dev/Settings/OnlineServicesPage.xaml.cs]: Can you add a try/catch on the result part? i.e. the one that gets a result from TryAuthenticateAsync

**Review by itsWindows11 — CHANGES_REQUESTED** [review]
Everything looks good, but try to add the suggestion.

File: Rise Media Player Dev/Settings/OnlineServicesPage.xaml.cs (hunk 1 above)
Comment (yourordinarycat) [reply Rise Media Player Dev/Settings/OnlineServicesPage.xaml.cs]: Added one for everything after getting the result (and when getting it), should be good with that.

**Review by itsWindows11 — APPROVED** [review]

**Pull request merged**